aiplatform/snippets/embeddings_preview.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START generativeaionvertexai_sdk_embedding]
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	aiplatformpb "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func If[T any](cond bool, vtrue, vfalse T) T {
+	if cond {
+		return vtrue
+	}
+	return vfalse
+}
+
+func embedTextsPreview(
+	apiEndpoint, project, model string, texts []string,
+	task string, dimensionality *int) ([][]float32, error) {
+	ctx := context.Background()
+
+	re := regexp.MustCompile(`(.+)-aiplatform.+`)
+	match := re.FindStringSubmatch(apiEndpoint)
+	location := "us-central1"
+	if match != nil {
+		location = match[1]
+	}
+
+	client, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	endpoint := fmt.Sprintf("projects/%s/locations/%s/publishers/google/models/%s", project, location, model)
+	instances := make([]*structpb.Value, len(texts))
+	for i, text := range texts {
+		instances[i] = structpb.NewStructValue(&structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				"content":   structpb.NewStringValue(text),
```

Comment:
"title" seems missing here

Reply:
Let's handle the title as a feature request, because other samples (nodejs, python, and java) are not processing the title.